### PR TITLE
Migrate node deletions to user-actions

### PR DESCRIPTION
**Pull request by viddo**
Milestone for #7881

This seem to have some unwanted consequences for the layer creations (tests failing at least), since the deletion is not handled implicitly anymore, ~~so might have to migrate those too as part of this~~ fixe by (temporarily) delete faulty analysis. Will be fixed in the next PR so don't want to waste time on that now, at least the layer creation can be migrated separately. 

@alonsogarciapablo can you review?

### CODE REVIEW

File: lib/assets/javascripts/cartodb3/data/user-actions.js [hunk 1]
```diff
@@ -88,7 +91,52 @@ module.exports = function (collections) {
...
+        // If deleted node was a head of a layer change the source to be its primary source instead
+        layerDefinitionsCollection.each(function (m) {
+          if (m.get('source') === nodeId) {
+            m.save({source: primarySourceNode.id});
+          }
+        });
+      }
+
+      var containsNode = function (m) {
+        return m.id !== nodeDefModel.id && m.containsNode(nodeDefModel);
```
Comment (alonsogarciapablo) [inline lib/assets/javascripts/cartodb3/data/user-actions.js]: `widgetModel.containsNode` and `layerDefinitionModel.containsNode` doesn't make a lot of sense naming wise IMO, but I like 👇 so 👍

Comment (viddo) [inline lib/assets/javascripts/cartodb3/data/user-actions.js]: for widget it's questionable yes, but for layer I believe it's accurate. :)

**Comment by alonsogarciapablo** [comment]
👍

**Comment by viddo** [comment]
Frontend tests passing, so merging

**Pull request merged**